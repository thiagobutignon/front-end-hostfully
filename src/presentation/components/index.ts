export { default as ErrorComponent } from './error/error'
export { default as HeaderComponent } from './header/header'
export { default as InputError } from './input/input-error'
export { default as InputWithLabel } from './input/input-with-label'
export { default as LayoutComponent } from './layout/layout'
export { default as PropertyInfoComponent } from './property-info/property-info'
export { default as PropertyCardComponent } from './select/property-select'
export { default as ThemeModeComponent } from './theme-mode/theme-mode'
