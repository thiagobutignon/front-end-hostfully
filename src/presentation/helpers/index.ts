export * from './populate-form-fields'
export * from './wrap-chakra'
