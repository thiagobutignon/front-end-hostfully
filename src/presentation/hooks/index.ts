export * from './booking'
export * from './error'
export * from './property'
