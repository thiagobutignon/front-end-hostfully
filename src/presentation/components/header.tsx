import { Box, Flex, Menu, Stack, Text } from '@chakra-ui/react'

import { ThemeModeComponent } from '@/presentation/components'

const HeaderComponent: React.FC = () => {
  return (
    <>
      <Box bg="reverseBackground" px={4}>
        <Flex h={16} alignItems={'center'} justifyContent={'space-between'}>
          <Text color="reverseText" fontSize="xl" fontWeight={700}>
            Hostfully
          </Text>

          <Flex alignItems={'center'}>
            <Stack direction={'row'}>
              <Menu>
                <ThemeModeComponent />
              </Menu>
            </Stack>
          </Flex>
        </Flex>
      </Box>
    </>
  )
}

export default HeaderComponent
