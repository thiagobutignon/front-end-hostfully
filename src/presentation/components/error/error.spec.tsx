import { fireEvent, render, screen } from '@testing-library/react'

import { ErrorComponent } from '@/presentation/components'

describe('ErrorComponent', () => {
  const errorMessage = 'An unexpected error occurred'
  const mockReload = jest.fn()

  beforeEach(() => {
    render(<ErrorComponent error={errorMessage} reload={mockReload} />)
  })

  test('renders the error message', () => {
    const errorText = screen.getByTestId('error')
    expect(errorText).toBeInTheDocument()
    expect(errorText).toHaveTextContent(errorMessage)
  })

  test('renders a button that can be clicked to reload', () => {
    const reloadButton = screen.getByTestId('reload')
    expect(reloadButton).toBeInTheDocument()
    fireEvent.click(reloadButton)
    expect(mockReload).toHaveBeenCalledTimes(1)
  })

  test('matches snapshot', () => {
    const { asFragment } = render(
      <ErrorComponent error={errorMessage} reload={mockReload} />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
