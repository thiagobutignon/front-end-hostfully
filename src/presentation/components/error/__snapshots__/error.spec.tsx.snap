// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorComponent matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="errorWrap css-xi606m"
  >
    <p
      class="chakra-text css-17u9iup"
      data-testid="error"
    >
      An unexpected error occurred
    </p>
    <button
      class="chakra-button css-kboz6o"
      data-testid="reload"
      type="button"
    >
      Try Again
    </button>
  </div>
</DocumentFragment>
`;
